src/app/app.ts

```diff
@@ -150,6 +150,51 @@ export function view(state: IState, actions: any) {
       id: 'app',
       class: classNames({'theme-dark': state.settings.darkTheme}),
       oncreate: (el: El) => actions.onCreate(el),
+      ondragover: (ev: Event) => {
+        ev.preventDefault()
+        ev.stopPropagation()
+      },
+      ondragleave: (ev: Event) => {
+        ev.preventDefault()
+        ev.stopPropagation()
+      },
+      ondrop: (ev: any, node: IGraphNodeData) => {
+        ev.preventDefault()
+        ev.stopPropagation()
+        for (const f of ev.dataTransfer.files) {
+          logger.debug('Dropped file path = ' + f.path)
+          if (f.path.includes('.txt')) {
+            const reader = new FileReader()
```

Comment:
Line 167 and 168 are repeated in on lines 182 and 183. We can remove code redundancy if we move these two lines outside the if-statement to line 165.